src/posix/client.cpp

```diff
@@ -296,44 +211,40 @@ int main(int argc, char *argv[])
             }
             else
             {
-                ssize_t lineStart = 0;
-
                 for (ssize_t i = 0; i < rval; i++)
                 {
-                    int prevPromptState = promptState;
+                    char c = buffer[i];
 
-                    if (FindPrompt(promptState, buffer[i]))
+                    lineBuffer[lineBufferWritePos++] = c;
+                    if (c == '\n' || lineBufferWritePos >= sizeof(lineBuffer) - 1)
                     {
-                        doneState  = 0;
-                        errorState = 0;
-                        lineStart  = i + 1;
-                        continue;
-                    }
-                    else if (prevPromptState == 1 && i == 0)
-                    {
-                        VerifyOrExit(DoWrite(STDOUT_FILENO, ">", 1), ret = OT_EXIT_FAILURE);
-                    }
+                        size_t skipPrefix = 0;
 
-                    if (buffer[i] == '\r' || buffer[i] == '\n')
-                    {
-                        VerifyOrExit(DoWrite(STDOUT_FILENO, buffer + lineStart, static_cast<size_t>(i - lineStart + 1)),
+                        // read one line successfully or line buffer is full
+                        lineBuffer[lineBufferWritePos] = '\0';
```

Comment:
Is this line necessary? It seems that each time to call `DoWrite`, this position won't be output. And things like `strlen` is not used here.

Reply:
Yes it is.
It's necessary for `strncmp`, for example in case the line in `lineBuffer` is shorter than `Done\r\n`